Makefile

```diff
@@ -1,14 +1,29 @@
-GOFLAGS :=
-IMAGE_REPOSITORY_NAME ?= openshift
+.PHONY: all
 
-build:
-	go build -mod=vendor $(GOFLAGS) .
-.PHONY: build
+# Include the library makefile
+include $(addprefix ./vendor/github.com/openshift/library-go/alpha-build-machinery/make/, \
+    golang.mk \
+    targets/openshift/deps-gomod.mk \
+    targets/openshift/images.mk \
+)
 
-images:
-	imagebuilder -f Dockerfile -t $(IMAGE_REPOSITORY_NAME)/oauth-proxy .
-.PHONY: images
+IMAGE_REGISTRY?=registry.svc.ci.openshift.org
+
+# This will call a macro called "build-image" which will generate image specific targets based on the parameters:
+# $0 - macro name
+# $1 - target name
+# $2 - image ref
+# $3 - Dockerfile path
+# $4 - context directory for image build
+# It will generate target "image-$(1)" for building the image and binding it as a prerequisite to target "images".
+$(call build-image,ocp-oauth-proxy,$(IMAGE_REGISTRY)/ocp/4.4:oauth-proxy,./Dockerfile.rhel7,.)
+
+# Run core verification and all self contained tests.
+check: | verify test-unit
```

Comment:
nit: a non-standard target, I don't object if you want it there

Reply:
I think I saw it somewhere else which was the sole reason I added it here, it shall be removed